src/lib.rs

```diff
@@ -1,5 +1,5 @@
 #![no_std]
-#![warn(missing_docs, unused_lifetimes)]
+#![warn(unused_lifetimes)]
```

Comment:
I think we should keep the `warn(missing_docs)` and add basic documentations to the new `_with_context` functions saying that they're the counterpart of the other functions, but with a context